lib/msal-angular/src/msal.interceptor.config.ts

```diff
@@ -3,10 +3,13 @@
  * Licensed under the MIT License.
  */
 
+import { HttpRequest } from "@angular/common/http";
 import { PopupRequest, RedirectRequest, InteractionType, SilentRequest } from "@azure/msal-browser";
 
+export declare type MsalInterceptorAuthRequest = Omit<PopupRequest, "scopes"> | Omit<RedirectRequest, "scopes"> | Omit<SilentRequest, "scopes"> ;
+
 export type MsalInterceptorConfiguration = {
     interactionType: InteractionType.Popup | InteractionType.Redirect;
     protectedResourceMap: Map<string, Array<string>>;
-    authRequest?: Omit<PopupRequest, "scopes"> | Omit<RedirectRequest, "scopes"> | Omit<SilentRequest, "scopes">;
+    authRequest?: MsalInterceptorAuthRequest | ((originalAuthRequest: MsalInterceptorAuthRequest, req: HttpRequest<unknown>) => MsalInterceptorAuthRequest);
```

Comment:
```suggestion
    authRequest?: MsalInterceptorAuthRequest | ((msalService: MsalService, req: HttpRequest<unknown>) => MsalInterceptorAuthRequest);
```

Reply:
The MSAL service doesn't provide exposed configuration. And the select account functionality is already build into the interceptor. 
This way it seems more like a way to override then to replace (what you are proposing).

Reply:
Currently the code in the interceptor is configured like this:
```js
const authRequest = typeof this.msalInterceptorConfig.authRequest === "function"
            ? this.msalInterceptorConfig.authRequest({ account: account }, req)
            : { ...this.msalInterceptorConfig.authRequest, account };
```

And just above that part is the logic to get to correct account.

That means the authRequest can be overriden like:
```js
authRequest: (authRequest, httpReq) => { 
  return { ...authRequest, authority: 'something here' };
}
```

And you don't have to copy the select the correct account logic. How about I add the MsalService as well, that way you can do both. So get the account in a none default way and just use the default way and override only the stuff you want overridden?